demisto_sdk/commands/test_content/test_modeling_rule/test_modeling_rule.py

```diff
@@ -318,7 +322,7 @@ def verify_results_against_test_data(
                 logger.debug(f"[cyan]{out}[/cyan]", extra={"markup": True})
                 result_test_case_system_out.append(out)
 
-                if type(result_val) == type(val) and result_val == val:
+                if result_val == val:
```

Comment:
the actual bug. comparing the types. there is an issue with int/float comparison of value returned from XSIAM. all numeric values are returned as float type.

Reply:
Maybe change the original check to something like:
introduce the following method (pseudocode,) and use it.
```
def get_type_family(object) -> Enum(STRING, NUMBER, DATE, BOOLEAN):
    if type(object) == StringType:
        return STRING
    if type(object) in [intType, floatType]:
        return NUMBER
    # same for DATE and BOOLEAN
```

Reply:
fixed.